pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamConfig.java

```diff
@@ -79,10 +78,11 @@ public class StreamConfig {
   // Allow overriding it to use different offset criteria
   private OffsetCriteria _offsetCriteria;
 
-  // Indicates if the segment should be uploaded to the deep store's file system or to the controller during the
-  // segment commit protocol. By default, segment is uploaded to the controller during commit.
-  // If this flag is set to true, the segment is uploaded to deep store.
-  private final boolean _serverUploadToDeepStore;
+  // Indicate StreamConfig flag for table if segment should be uploaded to the deep store's file system or to the
+  // controller during the segment commit protocol. if config is not present in Table StreamConfig
+  // _serverUploadToDeepStore is null and method isServerUploadToDeepStore() overrides the default value with Server
+  // level config
+  private final String _serverUploadToDeepStore;
```

Comment:
Let's make this an object `Boolean` to different explicit `false` and not set

Reply:
Fixed it.